### DISCUSSION TITLE
Link to math blocks

### DISCUSSION THREAD

**Discussion by patrick-egenlauf** [squidfunk/mkdocs-material#8035] [Q&A] [answered]
I am using the MathJax library to write math equations both in math blocks and inline. I would like to insert a link to an equation (block) in the text. Is this possible, and if not, would you consider adding this as a new feature?

**Comment by squidfunk** [comment]
You can always add a headline or `div` with an `id` around your block that you want to link to, or preface it with a headline. Other than that, directly linking to MathJax blocks is something that should be implemented by MathJax itself ☺️

``` md
<div id="my-anchor" markdown>

$$
\cos x=\sum_{k=0}^{\infty}\frac{(-1)^k}{(2k)!}x^{2k}
$$

</div>

**Reply by patrick-egenlauf** [reply]
Thanks for the quick reply. I tried it, but then the equation doesn't render properly (dollar signs and everything in between is displayed as plain text). 

I'm also not familiar with `html`, so how can I link to this block in my text, assuming it works?

**Reply by squidfunk** [reply]
Sorry, I forgot to mention, that you need to add the [`md_in_html`](https://python-markdown.github.io/extensions/md_in_html/) extension to `mkdocs.yml`

``` yaml
markdown_extensions:
  - md_in_html
```

**Comment by patrick-egenlauf** [comment, marked as answer]
This is already possible with MathJax via `\tag` and/or `\label` inside the equation. See for example the [MathJax documentation about automatic equation numbering](https://docs.mathjax.org/en/latest/input/tex/eqnumbers.html#automatic-equation-numbering) or [this stackexchange post](https://math.meta.stackexchange.com/questions/5020/mathjax-basic-tutorial-and-quick-reference/11491#11491) under *Tags & References* or *Equation numbering*.
